mods/ra/maps/allies-09a/allies09a-AI.lua

```diff
@@ -0,0 +1,88 @@
+--[[
+   Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+
+WTransWays =
+{
+	{ USSRRFEntry.Location, USSRUnload1.Location },
+	{ USSRRFEntry.Location, USSRUnload2.Location }
+}
+WTransUnits =
+{
+	hard = { { "3tnk", "3tnk", "3tnk", "v2rl", "v2rl" }, { "v2rl", "v2rl", "e4", "e4", "3tnk" } },
+	normal = { { "e1", "e1", "3tnk", "3tnk", "v2rl" }, { "e4", "e4", "e4", "e4", "v2rl" } },
+	easy = { { "e1", "e1", "e1", "e2", "e2" }, { "e2", "e2", "3tnk" } }
+}
+WTransDelays =
+{
+	easy = 9,
+	normal = 7,
+	hard = 5
+}
+BuildDelays =
+{
+	easy = 90,
+	normal = 60,
+	hard = 30
+}
+
+SovietAircraftType = { "yak" }
+Yaks = { }
+
+
+ProduceAircraft = function()
+	if Airfield.IsDead or Airfield.Owner ~= ussr then
+		return
+	end
+
+	ussr.Build(SovietAircraftType, function(units)
+		local yak = units[1]
+
+
+		Trigger.OnKilled(yak, ProduceAircraft)
+
+
+		InitializeAttackAircraft(yak, player)
+	end)
+end
+
+IdleHunt = function(unit) if not unit.IsDead then Trigger.OnIdle(unit, unit.Hunt) end end
+
+WTransWaves = function()
+	local way = Utils.Random(WTransWays)
+	local units = Utils.Random(WTransUnits)
+	local attackUnits = Reinforcements.ReinforceWithTransport(ussr, "lst", units , way, { way[2], way[1] })[2]
+	Utils.Do(attackUnits, function(a)
+		Trigger.OnAddedToWorld(a, function()
+			a.AttackMove(KosyginExtractPoint.Location)
+			IdleHunt(a)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(WTransDelays), WTransWaves)
+end
+
+ActivateAI = function()
+	local difficulty = Map.LobbyOption("difficulty")
+	WTransUnits = WTransUnits[difficulty]
+	WTransDelays = WTransDelays[difficulty]
+	BuildDelays = BuildDelays[difficulty]
+
+	local buildings = Utils.Where(Map.ActorsInWorld, function(self) return self.Owner == ussr and self.HasProperty("StartBuildingRepairs") end)
+	Utils.Do(buildings, function(actor)
+		Trigger.OnDamaged(actor, function(building)
+			if building.Owner == ussr and building.Health < building.MaxHealth * 3/4 then
+				building.StartBuildingRepairs()
+			end
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(1) + DateTime.Seconds(10), ProduceAircraft)
+	Trigger.AfterDelay(DateTime.Minutes(WTransDelays), WTransWaves)
+
+end
```

Comment:
Remove space

Reply:
Thank you for your review. Unused space remove. 
[4b4c38e](https://github.com/OpenRA/OpenRA/pull/18695/commits/4b4c38e8bed864fd0e1a1d749bc44fef1960b7b9)